src/common/transformations/src/transformations/common_optimizations/transpose_sinking_utils.cpp

```diff
@@ -71,12 +71,52 @@ void SwapNames(NodePtr node1, NodePtr node2) {
     SwapOutputNames(node1->output(0), node2->output(0));
 }
 
+namespace {
+
+bool HasDynamicRankInput(NodePtr node) {
+    for (size_t i = 0; i < node->get_input_size(); ++i) {
+        const auto input_node = node->input_value(i);
+        const ov::Rank output_rank = input_node.get_partial_shape().rank();
+        if (output_rank.is_dynamic())
+            return true;
+    }
+    return false;
+}
+
+ov::Rank::value_type GetMaxInputRank(NodePtr node) {
+    ov::Rank::value_type max_input_rank = 0;
+    for (size_t i = 0; i < node->get_input_size(); ++i) {
+        const auto input_node = node->input_value(i);
+        const ov::Rank output_rank = input_node.get_partial_shape().rank();
+        if (output_rank.is_dynamic())
+            return 0;
+        const ov::Rank::value_type output_rank_len = output_rank.get_length();
+        if (output_rank_len > max_input_rank)
+            max_input_rank = output_rank_len;
+    }
+    return max_input_rank;
+}
+
+NodePtr InsertUnsqueeze(Output<Node> node, size_t n_dims) {
+    std::vector<size_t> dims(n_dims);
+    std::iota(dims.begin(), dims.end(), 0);
+    auto unsqueeze_const = std::make_shared<Constant>(ov::element::u64, Shape{dims.size()}, dims);
+    auto unsqueeze = std::make_shared<Unsqueeze>(node, unsqueeze_const);
+    copy_runtime_info(node.get_node_shared_ptr(), {unsqueeze, unsqueeze_const});
```

Comment:
do we really need to copy runtime info here?

Reply:
This is needed at least since check_rt_info(function) throws exception